src/main/java/org/apache/pirk/schema/query/QuerySchemaLoader.java

```diff
@@ -246,8 +231,11 @@ private static QuerySchema loadQuerySchemaFile(String schemaFile, boolean hdfs,
     }
 
     // Create the query schema object
-    querySchema = new QuerySchema(schemaName, dataSchemaName, elementNames, selectorName, dataElementSize, filterNamesSet, filter);
 
+    DataFilter filter = instantiateFilter(filterTypeName, filterNamesSet);
+    querySchema = new QuerySchema(schemaName, dataSchemaName, selectorName, filterTypeName, filter, dataElementSize);
```

Comment:
+1 
Will go ahead and merge and we can address/not address the small item above as desired